src/actions.sol

```diff
@@ -185,21 +201,38 @@ contract Actions {
         returns (uint256 loan, uint256 tokenId)
     {
         tokenId = mintAsset(minter);
-        loan = issue(registry, tokenId);
+        loan = issue(shelf, registry, tokenId);
```

Comment:
I think it would be more elegant to implement internal functions like `_lock` and `_issue`. 

The public function performs the check `shelf_ == shelf` and here for example the internal function can be called.

Reply:
I agree, but I wasn't sure what was and wasn't internal, as everything but `_repay` is marked `public onlyDelegateCall`. Are `issue` and `_lock` the only other internal functions?